File: 2200. Find All K-Distant Indices in an Array/solution.java
class Solution {
  public List<Integer> findKDistantIndices(int[] nums, int key, int k) {
    List<Integer> keys = new ArrayList<Integer>();
    for (int index = 0; index < nums.length; index++){
      int num = nums[index];
      if (key == num){
        keys.add(index);
      }
    }

    int keyIndex = 0;
    List<Integer> result = new ArrayList<Integer>();
    for (int index = 0; index < nums.length; index++){
      if (keyIndex < keys.size()){
        if (Math.abs(index - keys.get(keyIndex)) <= k){
          result.add(index);
        }
        else if (keyIndex + 1 < keys.size()){
          if (Math.abs(index - keys.get(keyIndex + 1)) <= k){
            keyIndex += 1;
            result.add(index);
          }
        }
      }
    }

    return result;
  }
}

File: 2200. Find All K-Distant Indices in an Array/solution.js
/**
 * @param {number[]} nums
 * @param {number} key
 * @param {number} k
 * @return {number[]}
 */
 var findKDistantIndices = function(nums, key, k) {
  let keys = [];
  for (let index = 0; index < nums.length; index++){
    let num = nums[index];
    if (key === num){
      keys.push(index);
    }
  }

  let keyIndex = 0;
  let result = [];
  for (let index = 0; index < nums.length; index++){
    if (keyIndex < keys.length){
      if (Math.abs(index - keys[keyIndex]) <= k){
        result.push(index);
      }
      else if (keyIndex + 1 < keys.length){
        if (Math.abs(index - keys[keyIndex + 1]) <= k){
          keyIndex += 1;
          result.push(index);
        }
      }
    }
  }

  return result;
};